integration/shareddata/benchmarks.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shareddata
+
+import (
+	"fmt"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"golang.org/x/exp/maps"
+)
+
+// BenchmarkSmallDocuments provides 10000 documents that look like:
+//
+//	{_id: int32(0), id: int32(0), v: "foo"}
+//	{_id: int32(1), id: int32(1), v: int32(42)}
+//	{_id: int32(2), id: int32(2), v: "42"}
+//	{_id: int32(3), id: int32(3), v: {"foo": int32(42)}}
+//	...
+//
+// `_id` is a primary key that goes from 0 to n-1.
+// `id` has the same value as `_id`, but is not indexed by default.
+// `v` has one of the four values shown above.
+var BenchmarkSmallDocuments = newGeneratorBenchmarkProvider("SmallDocuments", 10000, func(n int) generatorFunc {
+	values := []any{
+		"foo", int32(42), "42", bson.D{{"foo", int32(42)}},
+	}
+	l := len(values)
+
+	var i int
+	return func() bson.D {
+		if i >= n {
+			return nil
+		}
+		doc := bson.D{
+			{"_id", int32(i)},
+			{"id", int32(i)},
+			{"v", values[i%l]},
+		}
+		i++
+		return doc
+	}
+})
+
+// BenchmarkLargeDocuments provides a single large document with fields of various types.
+var BenchmarkLargeDocuments = newGeneratorBenchmarkProvider("LargeDocuments", 123, func(n int) generatorFunc {
```

Comment:
We should make this one much better. For example, we could generate something similar to https://github.com/FerretDB/FerretDB/issues/2295#issuecomment-1488308844